plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -1671,15 +1669,15 @@ public static void suspendAll() {
                                 }
                             });
                         }
-                        cpsExec.getOwner().getListener().getLogger().close();
+                        if (cpsExec.owner != null) {
```

Comment:
main fix

Reply:
Any idea how this would be reachable? Corruption during resumption or something? Perhaps https://github.com/jenkinsci/workflow-api-plugin/pull/304 has made it possible? If I understand right, in this case `FlowExecutionList.runningTasks` contains a `FlowExecutionOwner` which successfully returns a `FlowExecution` from `.get()`, but for which `FlowExecution.owner` on the returned object is null, which seems problematic and unexpected.

Reply:
I am afraid I do not know—just saw this in a log.

Reply:
I suppose `[Workflow]Run.reload` deserializes `execution`, but then `WorkflowRun.onLoad` fails before calling `getExecution`, or `getExecution` fails before calling `fetchedExecution.onLoad(new Owner(this))`, or `unmarshal` fails partway through, etc. Presumably the build was badly corrupted somehow. The point of this PR is just to avoid unnecessary stack traces after that.

Reply:
For what it's worth though I saw another case of a `CpsFlowExecution` with a `null` `owner` recently, which is why I am wondering if something has changed things:

```
java.lang.IllegalStateException: List of flow heads unset for CpsFlowExecution[null]
	at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution.getCurrentHeads(CpsFlowExecution.java:1018)
        ... insignificant, just a user loading some build page that accessed the heads for an execution ...
```

I looked at the XML for that execution on disk and `head` was non-null and pointed to a `FlowEndNode` that did exist in `workflow/`, so it wasn't obvious what might have gone wrong.